### PR TITLE
NotFound RestEasy resources should not return any internal impl details

### PR DESCRIPTION
**Pull request by pjgg**
### Summary

Fails: 
```
cd http/rest-client
mvn clean verify -Dquarkus.platform.version=2.2.5.Final
```

Success: `mvn clean verify -Dall-modules -pl http/rest-client`

Please select the relevant options.

- [X] New scenario (non-breaking change which adds functionality)

### Checklist:
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: http/rest-client/src/main/java/io/quarkus/ts/http/restclient/RestInterface.java [hunk 1]
```diff
@@ -24,4 +24,9 @@ public interface RestInterface {
     @Path("/json")
     @Produces(MediaType.APPLICATION_JSON)
     Book getAsJson();
+
+    @GET
+    @Path("/notFound")
```
Comment (michalvavrik) [inline http/rest-client/src/main/java/io/quarkus/ts/http/restclient/RestInterface.java:29]: I don't see a point of exposing this endpoint when you never actually call it, please mention you are calling `given().get("/notFound")` not `given().get("/client/book/notFound")`.

**Review by michalvavrik — CHANGES_REQUESTED** [review]
Please answer my question before I merge request. Thanks

File: http/rest-client/src/main/java/io/quarkus/ts/http/restclient/RestInterface.java (hunk 1 above)
Comment (pjgg) [reply http/rest-client/src/main/java/io/quarkus/ts/http/restclient/RestInterface.java]: You are right!. Even I don't need an endpoint over `ClientBookResource`. The issue could be reproduced with a simple notFound.

**Review by michalvavrik — APPROVED** [review]
Lgtm, thank you.

**Pull request merged**